app/services/CompileService.scala

```diff
@@ -36,38 +35,40 @@ class CompileService @Inject()(
                                 actorSystem: ActorSystem
                               ) (implicit ec: ExecutionContext) {
 
-
   private def doCompile(currentVersionInfo: VersionInfo,
-                newVersionInfo: VersionInfo,
-                pstr: String,
-                userAnswers: UserAnswers,
-                delete :Boolean,
-                compileResponse: Future[Unit])(implicit headerCarrier: HeaderCarrier): Future[Unit] = {
+                        newVersionInfo: VersionInfo,
+                        pstr: String,
+                        userAnswers: UserAnswers,
+                        delete: Boolean,
+                        eventType: Option[EventType],
+                        deleteAttr: Option[(String, EventDataIdentifier, Int, String)])(implicit headerCarrier: HeaderCarrier): Future[Unit] = {
```

Comment:
Instead of using two options for eventType and deleteAttr, can we use Either[EventType, (String, EventDataIdentifier, Int, String)] instead?

This would avoid user error from a developer not using the function correctly.

Reply:
Sure will try that